### ISSUE TITLE
Automatic translation for removing Shadow DOM selectors lowers specificity

### ISSUE THREAD

**Issue by simurai** [atom/atom#13019]
Carried over from https://github.com/lee-dohm/red-wavy-underline/issues/10 since it's more a core issue.
### Prerequisites
- [ ] Can you reproduce the problem in [safe mode](http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode)?
- [x] Are you running the [latest version of Atom](http://flight-manual.atom.io/hacking-atom/sections/debugging/#update-to-the-latest-version)?
- [x] Did you check the [debugging guide](http://flight-manual.atom.io/hacking-atom/sections/debugging/)?
- [x] Did you check the [FAQs on Discuss](https://discuss.atom.io/c/faq)?
- [x] Are you reporting to the [correct repository](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#atom-and-packages)?
- [x] Did you [perform a cursory search](https://github.com/issues?q=is%3Aissue+user%3Aatom+-repo%3Aatom%2Felectron) to see if your bug or enhancement is already reported?

For more information on how to write a good [bug report](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#how-do-i-submit-a-good-bug-report) or [enhancement request](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#how-do-i-submit-a-good-enhancement-suggestion), see the `CONTRIBUTING` guide.
### Description

The automatic translation for removing Shadow DOM selectors lowers specificity. This can cause some packages to break.
### Steps to Reproduce
1. Checkout recent master (`1.13.0-dev-602af52` or later)
2. `apm install red-wavy-underline@0.3.0`
3. Make sure you have the [spell-check](https://atom.io/packages/spell-check) package enabled.
4. Open a file and misspell a word. Like `mispell`.

**Expected behavior:**

There should only be the red wavy underline.

<img width="179" alt="stable_360" src="https://cloud.githubusercontent.com/assets/378023/19508805/39dd2976-9616-11e6-8d3f-545c9d46517f.png">

**Actual behavior:**

There are the red wavy underline **and** the default dashed underline.

<img width="180" alt="shadow-dom-changes_360" src="https://cloud.githubusercontent.com/assets/378023/19508799/2fc664e8-9616-11e6-849d-cfe9864dfd87.png">
### Versions

You can get this information from executing `atom --version` and `apm --version` at the command line. Also, please include the OS and what version of the OS you're running.

```
macOS 10.12 (16A323)

Atom    : 1.13.0-dev-602af52
Electron: 1.3.6
Chrome  : 52.0.2743.82
Node    : 6.3.0

apm  1.13.0
npm  3.10.5
node 4.4.5
python 2.7.10
git 2.9.1
```
### Affected packages
- [x] [red-wavy-underline](https://github.com/lee-dohm/red-wavy-underline/issues/10)

**Comment by simurai** [comment]
Ok, I think the problem is this:

With Shadow DOM | Without Shadow DOM
--- | ---
![screen shot 2016-10-19 at 2 39 06 pm](https://cloud.githubusercontent.com/assets/378023/19507725/d5854180-960f-11e6-9f30-a66eac845260.png) | ![screen shot 2016-10-19 at 3 14 23 pm](https://cloud.githubusercontent.com/assets/378023/19507724/d584c502-960f-11e6-8bcc-20ebe3d1c2b5.png)

- With Shadow DOM, the selector of that package `atom-text-editor::shadow .misspelling` has a **higher** specificity than `.spell-check-misspelling .region`.
- Without Shadow DOM, the selector of that package `atom-text-editor .misspelling` has a **lower** specificity than `.spell-check-misspelling .region`.

So `border-bottom: none;` has no effect.

This is a bit odd, because pseudo selectors actually shouldn't be stronger than a class, but I guess `::shadow` has some special strengths?

## Possible solutions:

1. Do nothing. Some packages will break and the package author will have to fix the selector. We should add a note to the [docs](https://github.com/atom/flight-manual.atom.io/pull/281) though.
2. Do an automatic translation of `atom-text-editor::shadow` -> `atom-text-editor.editor` to increase specificity. I think `.editor` is deprecated, but currently still gets added to all text editors.

Not sure how widespread this problem is. I guess most selectors should be fine, because people probably copy the whole selector in the DevTools `.spell-check-misspelling .region` and just add `atom-text-editor::shadow` in front ->  `atom-text-editor::shadow .spell-check-misspelling .region` which then is still stronger even without `::shadow`.

/cc @as-cii

**Comment by simurai** [comment]
Chatted with @as-cii. We're gonna wait for a bit and see how big of an impact issue has before deciding what to do. Feel free to comment here if you find other affected packages.

**Comment by lee-dohm** [comment]
I fixed red-wavy-underline and published it requiring at least Atom v1.13.0, so I updated the repro steps above to install the version that is broken.
